docs/rules/0123/resource-pattern-singular.md

```diff
@@ -0,0 +1,85 @@
+---
+rule:
+  aip: 123
+  name: [core, '0123', resource-pattern-singular]
+  summary: Resource patterns must use the singular as the resource ID segment
+permalink: /123/resource-pattern-singular
+redirect_from:
+  - /0123/resource-pattern-singular
+---
+
+# Resource `pattern` use of `singular`
+
+This rule enforces that messages that have a `google.api.resource` annotation
+use the `singular` form as the resource ID segment, as described in [AIP-123][].
+
+## Details
+
+This rule scans messages with a `google.api.resource` annotation, and validates
+the `singular` form of the resource type name is used as the resource ID segment
+in every pattern.
+
+**Note:** Special consideration is given to type names of child collections that
+stutter next to their parent collection, as described in
+[AIP-122 Nested Collections][nested]. See AIP-122 for more details.
+
+**Important:** Do not accept the suggestion if it would produce a backwards
+incompatible change.
+
+## Examples
+
+**Incorrect** code for this rule:
+
+```proto
+// Incorrect.
+message Book {
+  option (google.api.resource) = {
+    type: "library.googleapis.com/BookShelf"
+    // resource ID segment doesn't match the singular.
+    pattern: "publishers/{publisher}/bookShelves/{shelf}"
+    singular: "bookShelf",
+  };
+
+  string name = 1;
+}
+```
+
+**Correct** code for this rule:
+
+```proto
+// Correct.
+message Book {
+  option (google.api.resource) = {
+    type: "library.googleapis.com/BookShelf"
+    pattern: "publishers/{publisher}/bookShelves/{book_shelf}"
+    singular: "bookShelf",
```

Comment:
Should we include the plural: "bookShelves" here as well? If I understand correctly the linter will fail if that doesn't exist as it can't go from singular --> plural

Reply:
You're right we shouldn't have incomplete examples. Done!